internal/services/cdn/cdn_endpoint_custom_domain_resource_test.go

```diff
@@ -30,20 +32,49 @@ func preCheck(t *testing.T) {
 	}
 }
 
+func preCheckUserManagedCertificate(t *testing.T) {
+	preCheck(t)
+	variables := []string{
+		// PFX File
+		"ARM_TEST_DNS_CERTIFICATE",
+
+		// Subdomain Name, this needs to be the same value as set to "CN" in the certificate.
+		"ARM_TEST_DNS_SUBDOMAIN_NAME",
+	}
+
+	for _, variable := range variables {
+		value := os.Getenv(variable)
+		if value == "" {
+			t.Skipf("`%s` must be set for acceptance tests!", variable)
+		}
+	}
+}
+
 type CdnEndpointCustomDomainResource struct {
 	DNSZoneRG     string
 	DNSZoneName   string
 	SubDomainName string
+	Certificate   string
 }
 
-func NewCdnEndpointCustomDomainResource(dnsZoneRg, dnsZoneName string) CdnEndpointCustomDomainResource {
-	return CdnEndpointCustomDomainResource{
+func NewCdnEndpointCustomDomainResource(dnsZoneRg, dnsZoneName string) *CdnEndpointCustomDomainResource {
+	return &CdnEndpointCustomDomainResource{
 		DNSZoneRG:     dnsZoneRg,
 		DNSZoneName:   dnsZoneName,
 		SubDomainName: acceptance.RandString(3),
 	}
 }
 
+func (r *CdnEndpointCustomDomainResource) WithCertificate(cert string) *CdnEndpointCustomDomainResource {
+	r.Certificate = cert
+	return r
+}
+
+func (r *CdnEndpointCustomDomainResource) WithSubDomain(subDomain string) *CdnEndpointCustomDomainResource {
+	r.SubDomainName = subDomain
+	return r
+}
```

Comment:
we're intentionally avoiding Fluent syntax within the provider since it's not used in Go fwiw:

```suggestion
func NewCdnEndpointCustomDomainResource(dnsZoneRg, dnsZoneName string) CdnEndpointCustomDomainResource {
	return CdnEndpointCustomDomainResource{
		DNSZoneRG:     dnsZoneRg,
		DNSZoneName:   dnsZoneName,
		SubDomainName: acceptance.RandString(3),
	}
}
```

since the tests have access to these properties, we could instead set them directly?

> r := NewCdnEndpointCustomDomainResource(os.Getenv("ARM_TEST_DNS_ZONE_RESOURCE_GROUP_NAME"), os.Getenv("ARM_TEST_DNS_ZONE_NAME"))
> r.Certificate = "something"